core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java

```diff
@@ -168,11 +168,14 @@ private UnderFileSystem getOrAddWithRecorder(AlluxioURI ufsUri,
       mCloser.register(fs);
       try {
         connectUfs(fs);
-      } catch (IOException e) {
+        tryUseFileSystem(fs, ufsUri.getPath());
+      } catch (Exception e) {
+        mUnderFileSystemMap.remove(key);
         String message = String.format(
             "Failed to perform initial connect to UFS %s: %s", ufsUri, e);
         recorder.record(message);
         LOG.warn(message);
+        throw new RuntimeException(e);
```

Comment:
Would you mind explaining why you want to throw the exception after catching it?

Reply:
If don't throw the exception, the created ufs client will be saved by `mMountIdToUfsInfoMap`, but this client may not be available. It affects the operations of the mount point.

Reply:
Thanks for the clarification.